src/main/java/de/uni_stuttgart/it_rex/course/repository/written/CourseRepository.java

```diff
@@ -11,5 +13,5 @@
  */
 @SuppressWarnings("unused")
 @Repository
-public interface CourseRepository extends JpaRepository<Course, UUID> {
-}
+public interface CourseRepository extends JpaRepository<Course, UUID>,
```

Comment:
Wasn das?

Reply:
JpaSpecificationExecutor<Course> sorgt dafür, dass das Repository findAll(Specification<Course> spec) ausführen kann, also dass man nach Specifications suchen kann. Dafür muss man das Interface nur extenden.

Wird dann benutzt in CourseService.java

Die Idee ist, dass man eine Specification erzeugt aus einem gegebenen Beispiel und Optional dem Zusatz, dass das End-Datum in der Zukunft liegt. Die Specification ist im Prinzip eine Konjunktion von Predicates.

Reply:
nice